### ISSUE TITLE
Failure 1020 only on iOS

### ISSUE THREAD

**Issue by Maxpower7847** [bigbluebutton/bigbluebutton#9339]
Hello, first of I am sorry if this is in the wrong subsection, but it seemed most fitting here.
We are currently in the process of deploying bbb at a university and I encountered an -to me- confusing problem: 
Our plan is to have seminars hosted through the university wifi. I gave the required ports to the IT-department and they released them which i could confirm since I am now able to connect to the bbb server from my laptop or my android device which are connected tothe university wifi. 

however the webcam doesn't work within iOS. I can enter a meeting but as soon as I want to start a webcamstream I get a 1020 error. 
From what I understand from similar posts, this usually indicates a port problem, but that wouldnt explain why the problem only occurs in IOS would it ? 
The iPhone used for testing is on the current iOS version and connects to the server just fine when it is disconnected from the wifi. 
In conclusion I am confused where the Problem is located. 
Doesn't really make sense that it's the bbb server since the other devices work. 
Doesn't really make sense that it's the firewall since the other devices work.
Doesn't really make sense that it's iOS since it works from other networks. 

So I hope someone can point me in a direction as to how to solve this Problem  

thank you in advance

**Comment by basisbit** [comment]
Is the bbb server set up to only have IPv4? I heard of problems regarding having IPv4 and IPv6 set up for the bbb server. Also, are you using an IPv4-only TURN server?

**Comment by Maxpower7847** [comment]
first thank you for your reply, I am very new to Server configurations in genera so I am sorry if I misunderstand something. 

iPv6/v4: when i use "ip addr | grep inet6" the first line i receive back is "inet6 ::1/128 scope host". my understanding is that means the server is configured for ipv6 as well correct ?

I am currently not using any TURN-Server.

**Comment by Maxpower7847** [comment]
Ok so we were able to somewhat limit the cause of the problem. 
Aparrently bbb is trying to connect to a STUN-Server over port 3478. 
This is kind of weird since there shouldn't be any STUN(or TURN)-Server-connection necessary:
The devices connecting are all in the same network. The other required ports are all opened as suggested in the documentation and there is no NATing going on, all clients have a public IP-Adress. 
Also this still doesn't really explain why the problems only occur on IOS-devices. 

I wanted to deactivate the STUN-Server completly as described in the ninth post here:  https://groups.google.com/forum/#!topic/bigbluebutton-setup/0A_D27r4jIw

since the bigbluebuttonversion described in this post is different I edited the file /usr/share/bbb-web/WEB-INF/classes/spring/turn-stun-servers.xml instead (commenting out the three described lines of code. 

The result of that is, that after restarting the server I know get an errormessage saying "Invalid BigBlueButton Endpoint and Secret" when trying to login in greenlight. 

I am a bit confused anyways. Why is bbb trying to use a stunserver when there is no nating going on in the first place ? do i have to change a different file to deactivate it ?

**Comment by rasos** [comment]
We have a working Stun / TURN server setup. The iPhone is NOT sending camera (issuing a 1020 error) ONLY when connected to a restricted WiFi (ports 443, 80 only), but it does receive screenshares and cameras from other participants. As soon as the iPhone switches to an unrestricted WiFi, also its camera works fine.

**Comment by prlanzarin** [comment]
Merging into https://github.com/bigbluebutton/bigbluebutton/issues/10746
